packages/components/src/Notification/ToastNotification/context/ToastNotificationContext.tsx

```diff
@@ -44,18 +44,23 @@ export const ToastNotificationProvider = ({
   const [notifications, setNotifications] = useState<ToastNotificationObj[]>([])
 
   const addToastNotification: ToastNotificationContextValue["addToastNotification"] =
-    notification => {
-      const uuid = uuidv4()
-      const notificationWithId = { id: uuid, ...notification }
-
-      const notificationExists = notifications.find(
-        ({ id }) => id === notification.id
-      )
-
-      if (!notificationExists) {
-        setNotifications(existing => [...existing, notificationWithId])
-      }
-    }
+    useCallback(
+      notification => {
+        const uuid = uuidv4()
+        const notificationWithId = { id: uuid, ...notification }
+
+        setNotifications(existing => {
+          const notificationExists = existing.find(
```

Comment:
Given the problem statement where you mention that checking the existing notifications isn't working, I'm surprised that the solution isn't something within this `existing.find` check but instead is the whole `addToastNotification`.

I'm not against the fix itself if it works to be able to unblock you, but it sounds like it might be in the wrong place.

Reply:
We found that we need to use the previous state syntax of setting state, because at the time that the check was being made, the notifications array hadn't been updated yet. By using `setNotifications( existing () => ...)` we are waiting for the previous state to have been set